src/receiver.rs

```diff
@@ -195,6 +200,11 @@ fn receive_packets<T>(recv: PacketReceiver, kill: Arc<AtomicBool>, send: Sender<
         trace!("Waiting on packet at {}...", recv);
         let (msg_bytes, addr) = match recv.recv_pckt() {
             Ok((bytes, addr)) => (bytes, addr),
+            Err(ref err) if err.kind() == io::ErrorKind::WouldBlock => {
```

Comment:
Looks like there was: https://github.com/rust-lang/rust/issues/27773#issuecomment-131196243
And in fact it's mentioned and documented: https://doc.rust-lang.org/stable/std/net/struct.UdpSocket.html#method.set_read_timeout